source/server-discovery-and-monitoring/tests/integration/closeInUse-pool-clear.yml

```diff
@@ -0,0 +1,108 @@
+# Test connection pool clear(closeInUse:true).
+runOn:
+    # This test requires failCommand appName, and
+    # failCommand blockConnection with closeConnection:true (SERVER-53512).
+    - minServerVersion: "4.9"
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "closeInUse"
+
+data: []
+
+tests:
+  - description: pool clear close in use true generates expected events
+    useMultipleMongoses: false,
+    clientOptions:
+      connectTimeoutMS: 500, 
+      heartbeatFrequencyMS: 500
+      appname: "closeInUse"
+      retryReads: false
```

Comment:
Can we add a similar test that has `retryReads: true` and verifies that the `find` succeeds? We can call it something like "Error returned from connection pool clear with closeInUseConnections=true is retryable".

Reply:
working on this, for some reason it became flaky on certain .net target frameworks, will provide this test separately

Reply:
done